File: benchmarks/java/integer_boolean/Java_Big_Integer/Or.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;

public class Or {






    private BInteger counter;

    public Or() {
        counter = (BInteger) new BInteger("0");
    }

    public void simulate() {
        while((counter.less(new BInteger("5000000")).or(new BInteger("1").equal(new BInteger("2")))).booleanValue()) {
            counter = (BInteger) counter.plus(new BInteger("1"));
        }
    }

    public static void main(String[] args) {
        Or exec = new Or();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}

File: microbenchmarks/cpp/integer_boolean/big_integer/Or.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef Or_H
#define Or_H

using namespace std;

class Or {



    private:



        BInteger counter;

    public:

        Or() {
            counter = (BInteger("0"));
        }

        void simulate() {
            while((counter.less((BInteger("5000000")))._or((BInteger("1")).equal((BInteger("2"))))).booleanValue()) {
                counter = counter.plus((BInteger("1")));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    Or exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

